source/Ship.h

```diff
@@ -503,16 +505,19 @@ class Ship : public Body, public std::enable_shared_from_this<Ship> {
 	bool isThrusting = false;
 	bool isReversing = false;
 	bool isSteering = false;
+	bool firePrimary = false;
```

Comment:
if I see it right this variable always equals this: firingCommands.Has(Command::PRIMARY)
so we do not need the setter?

Reply:
To give some context about how this applies for animation, the `firePrimary` bool essentially indicates to the Ship that the player wants to fire, and that the ship should enter the FIRING animation state.

Commands were initially really confusing to me, but this is how they seem to work.
In ship there are two sets of commands: `FireCommand firingCommands` and `Command commands`

`firingCommands` is simply a set of bits which indicate whether a certain weapon should fire (e.g. if the bit at index 4 is set to 1, then the weapon at index 4 should fire in this step). For beam weapons, this bit is set to 1 at all times, but for projectile/blaster type weapons, the bit oscillates between 0 and 1 depending on whether the weapon should fire in this frame. `firingCommands` also does not have any indication whether Command::PRIMARY or for that matter any of the Command constants are present or not.

`commands` was where I initially thought that directives like Command::PRIMARY would exist, however, `commands` only handles the non-weapon aspects of controlling a ship (e.g. Command::DEPLOY, Command::JUMP, etc.)

Where Command::PRIMARY is actually set is in AI.cpp in the `activeCommands` variable which is invisible to Ship. The  firePrimary bool essentially indicates whether Command::PRIMARY is set or not.

I will note that it is somewhat of a haphazard solution, a cleaner way of doing this would be add all of the activeCommands to the commands sent to the ship. But, I want to hear what you and @Hurleveur have to say before I do that implementation.

Reply:
I think its fine this way honestly but if you feel like its cleaner dont let me stop you. It all depends on how many variables this would save, however it would come at the cost of extra dependency between the two classes.